rest-service/manager_rest/deployment_update/step_extractor.py

```diff
@@ -177,24 +252,111 @@ def extract_steps(self):
         old = self.old_deployment_plan
         new = self.new_deployment_plan
 
+        self._extract_steps_from_description(old[DESCRIPTION],
+                                             new[DESCRIPTION])
+
+        self._extract_step_host_agent_plugins_steps(
+            old[NODES],
+            new[NODES]
+        )
+
+        self._extract_central_deployment_agent_plugins_steps(
+            old[DEPLOYMENT_PLUGINS_TO_INSTALL],
+            new[DEPLOYMENT_PLUGINS_TO_INSTALL])
+
         self._extract_steps(new, old)
         self._inverted_diff_perspective = True
         self._extract_steps(old, new)
 
-        return self.steps
+        supported_steps = [step for step in self.steps if step.supported]
+        supported_steps.sort()
+
+        unsupported_steps = [step for step in self.steps if not step.supported]
+
+        return supported_steps, unsupported_steps
+
+    def _extract_steps_from_description(self,
+                                        old_description,
+                                        new_description):
+        with self.entity_id_builder.extend_id(DESCRIPTION):
+            if old_description is None:
+                if new_description is not None:
+                    self._create_step(DESCRIPTION)
+            else:
+                if new_description is None:
+                    self._inverted_diff_perspective = True
+                    self._create_step(DESCRIPTION)
+                    self._inverted_diff_perspective = False
+                else:
+                    if old_description != new_description:
+                        self._create_step(DESCRIPTION, modify=True)
+
+    def _extract_step_host_agent_plugins_steps(self, old_nodes, new_nodes):
+        with self.entity_id_builder.extend_id(HOST_AGENT_PLUGINS):
+            for new_node_name in new_nodes:
+                new_node = new_nodes[new_node_name]
+                new_plugins_to_install = new_node[PLUGINS_TO_INSTALL]
+                if new_plugins_to_install:
+                    if new_node_name in old_nodes:
+                        with self.entity_id_builder.extend_id(new_node_name):
+                            old_node = old_nodes[new_node_name]
+                            if old_node != new_node:
+                                old_plugins_to_install = old_node[
+                                    PLUGINS_TO_INSTALL]
+                                for new_plugin_to_install in \
```

Comment:
Ps = [p for p in new_plugins_to_install if p['install'] and p not in old_plugins_to_install]